hack/graph.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+#
+# Usage:
+#
+#   graph.sh [URI] >graph.dot
+#
+# For example:
+#
+#   graph.sh https://cincinnati.example.org/your/graph | dot -Tsvg >graph.svg
+set -e
+
+URI="${1:-https://api.openshift.com/api/upgrades_info/v1/graph?channel=prerelease-4.1}"
+JQ_SCRIPT="$(cat <<EOF
+  "digraph Upgrades {\n  labelloc=t;\n  rankdir=BT;" as \$header |
+  (
+    [
+      .nodes |
+      to_entries[] |
+      "  " + (.key | tostring) +
+             " [ label=\"" + .value.version + "\"" + (
+               if .value.metadata.url then " href=\"" + .value.metadata.url + "\\"" else "" end
+             ) +
+             " ];"
+    ] | join("\n")
+  ) as \$nodes |
+  (
+    [
+      .edges[] |
+      "  " + (.[0] | tostring) + "->" + (.[1] | tostring) + ";"
+    ] | join("\n")
+  ) as \$edges |
+  [\$header, \$nodes, \$edges, "}"] | join("\n")
+EOF
+)"
+curl --silent --header 'Accept:application/json' "${URI}" | jq -r "${JQ_SCRIPT}"
```

Comment:
I would decrease the scope of this program and make it expect the content on stdin, leaving the fetching of the graph to the caller.

Reply:
> I would decrease the scope of this program and make it expect the content on stdin...

I think it's a bit tedious to have to remind myself of "the usual" URI and set the `Accept` header.  But I don't feel *that* strongly about it ;).  Done with a3bb428 -> 7ec4253.